### PR TITLE
Account

### LINKED ISSUE #1: Implementation of account management
This section includes adding, editing, deleting and searching accounts

### LINKED ISSUE #4: Implementation of Transcation management
This section includes transfer, update, rollback

### PR DESCRIPTION
**Pull request by HamidNiakan**

### CODE REVIEW

File: src/Http/Controllers/AccountController.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+use dnj\Account\Http\Requests\CreateNewAccountRequest;
+use dnj\Account\Http\Resources\AccountResource;
+use dnj\Number\Number;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Http\Request;
+
+class AccountController extends Controller {
+	public AccountManager $account_manager;
+	
+	public function __construct ( AccountManager $account_manager ) {
```
Comment (yeganemehr) [inline src/Http/Controllers/AccountController.php]:
```suggestion
	public function __construct ( protected AccountManager $accountManager ) {
```

We use php 8.1, you can code less.
for naming variables use camelCase

File: src/Http/Controllers/AccountController.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+		$this->account_manager = $account_manager;
+	}
+	
+	public function create ( CreateNewAccountRequest $request ) {
+		if ( !auth()->check() ) {
```
Comment (yeganemehr) [inline src/Http/Controllers/AccountController.php]: Do it as "auth" middleware in routes

File: src/Http/Controllers/AccountController.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+			abort(401 , 'unauthenticated');
+		}
+		$title = $request->get('title');
+		$currency_id = $request->get('currency_id');
+		$can_send = $request->get('can_send');
+		$can_receive = $request->get('can_receive');
+		$meta = $request->get('meta');
+		$user_id = auth()->user() ? auth()->user()->id : null;
+		$account = $this->account_manager->create($title , $currency_id , $user_id , AccountStatus::ACTIVE , $can_send , $can_receive , $meta);
+		$account = AccountResource::make($account);
```
Comment (yeganemehr) [inline src/Http/Controllers/AccountController.php]: just return the resource

File: src/routes/api.php [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+use Illuminate\Support\Facades\Route;
+
+/*
```
Comment (yeganemehr) [inline src/routes/api.php]: Remove educational comments.

File: src/routes/api.php [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+|--------------------------------------------------------------------------
+|
+| Here is where you can register API routes for your application. These
+| routes are loaded by the RouteServiceProvider within a group which
+| is assigned the "api" middleware group. Enjoy building your API!
+|
+*/
+Route::controller(\dnj\Account\Http\Controllers\AccountController::class)
+	 ->group(function () {
+		 Route::post('create' , 'create');
```
Comment (yeganemehr) [inline src/routes/api.php]: Read about RESTFUL url template
search => GET /accounts
create => POST /accounts
update => PUT /accounts/{accountId}
delete => DELETE /accounts/{accountId}

**Review by yeganemehr — CHANGES_REQUESTED** [review]

File: src/Http/Controllers/AccountController.php [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+use dnj\Account\Models\Account;
+use Illuminate\Support\Str;
+
+class AccountController extends Controller
+{
+    public function __construct(protected AccountManager $accountManager)
+    {
+    }
+
+    public function index(AccountSearchRequest $request)
```
Comment (yeganemehr) [inline src/Http/Controllers/AccountController.php:20]: please test this function

File: src/Http/Controllers/TransactionController.php [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+use dnj\Account\TransactionManager;
+use dnj\Number\Number;
+
+class TransactionController extends Controller
+{
+    public function __construct(protected TransactionManager $transactionManager)
+    {
+    }
+
+    public function index(Account $account, TransactionSearchRequest $request)
```
Comment (yeganemehr) [inline src/Http/Controllers/TransactionController.php:20]: please test this method.

**Review by yeganemehr — CHANGES_REQUESTED** [review]

File: src/Http/Controllers/AccountController.php (hunk 6 above)
Comment (HamidNiakan) [reply src/Http/Controllers/AccountController.php:20]: done

File: src/Http/Controllers/TransactionController.php (hunk 7 above)
Comment (HamidNiakan) [reply src/Http/Controllers/TransactionController.php:20]: done

**Review by yeganemehr — APPROVED** [review]

**Pull request merged**